File: 2018/Jul15/Troy/Q2.java
class Solution {
    public int dominantIndex(int[] nums) {
        int max = maxInt(nums);
        int index = 0;
        for(int i = 0; i < nums.length; i++) {
            if (nums[i] == max) {
                index = i;
            } else if ((2 * nums[i]) > max) {
                return -1;
            }
        }
        return index;
    }
    public int maxInt(int[] nums) {
        int result = 0;
        for(int i = 0; i < nums.length; i++) {
            if (nums[i] > result) result = nums[i];
        }
        return result;
    }
}

File: 2018/Jul15/Carl/Q2.py
class Solution(object):
    def dominantIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return -1
        elif len(nums) == 1:
            return 0
        maxValue = 0
        secMaxValue = 0
        maxIndex = 0
        secMaxIndex = 0
        for i, value in enumerate(nums):
            if value > maxValue:
                print(value)
                secMaxIndex = maxIndex
                secMaxValue = maxValue
                maxIndex = i
                maxValue = value
            elif value > secMaxValue:
                secMaxIndex = i
                secMaxValue = value
        print("mv: %d, smv: %d, mi: %d, smi: %d" %
              (maxValue, secMaxValue, maxIndex, secMaxIndex))
        if maxValue >= 2 * secMaxValue:
            return maxIndex
        elif secMaxValue >= 2 * maxValue:
            return secMaxIndex
        else:
            return -1


s = Solution()
print(s.dominantIndex([2, 0, 1, 1]))
